go-controller/pkg/ovn/loadbalancer.go

```diff
@@ -230,39 +229,38 @@ func (ovn *Controller) createLoadBalancerRejectACL(lb string, sourceIP string, s
 		klog.Errorf("Error while querying ACLs by name: %s, %v", stderr, err)
 	} else if len(aclUUID) > 0 {
 		klog.Infof("Existing Service Reject ACL found: %s for %s", aclUUID, aclName)
-		// If we found the ACL exists we need to ensure it applies to all logical switches
-		cmd := []string{}
-		for _, ls := range switches {
-			cmd = append(cmd, "--", "add", "logical_switch", ls, "acls", aclUUID)
-		}
-		if len(cmd) > 0 {
-			_, _, err = util.RunOVNNbctl(cmd...)
-			if err != nil {
-				klog.Warningf("Unable to add reject ACL: %s for switches: %s", aclUUID, switches)
-			}
+		_, stderr, err = util.RunOVNNbctl("--", "add", "port_group", ovn.clusterPortGroupUUID, "acls", aclUUID)
+		if err != nil {
+			klog.Errorf("Failed to add LB %s ACL %s %q to cluster port group: %q (%v)",
+				lb, aclUUID, aclName, stderr, err)
+			return "", err
 		}
+
 		ovn.setServiceACLToLB(lb, vip, aclUUID)
+
+		// If reject ACL exist, ensures that the _uuid is removed from logical_switch acls list.
+		// This step is required to ensure the clean-up when ovn upgrades from logical_switch acls
+		// to port_group based acls.
+		ovn.removeACLFromNodeSwitches(switches, aclUUID)
 		return aclUUID, nil
 	}
 
 	aclMatch = fmt.Sprintf("match=\"%s.dst==%s && %s && %s.dst==%d\"", l3Prefix, sourceIP,
 		strings.ToLower(string(proto)), strings.ToLower(string(proto)), sourcePort)
-
-	cmd := []string{"--id=@acl", "create", "acl", "direction=from-lport", "priority=1000", aclMatch, "action=reject",
+	cmd := []string{"--id=@reject-acl", "create", "acl", "direction=from-lport", "priority=1000", aclMatch, "action=reject",
 		fmt.Sprintf("name=%s", aclName)}
-	for _, ls := range switches {
-		cmd = append(cmd, "--", "add", "logical_switch", ls, "acls", "@acl")
-	}
-
+	cmd = append(cmd, "--", "add", "port_group", ovn.clusterPortGroupUUID, "acls", "@reject-acl")
 	aclUUID, stderr, err = util.RunOVNNbctl(cmd...)
 	if err != nil {
-		return "", fmt.Errorf("error creating ACL reject rule: %s for load balancer %s: %s, %s", cmd, lb, stderr,
-			err)
-	} else {
-		// Associate ACL UUID with load balancer and ip+port so we can remove this ACL if
-		// backends are re-added.
-		ovn.setServiceACLToLB(lb, vip, aclUUID)
+		klog.Errorf("Failed to add LB %s ACL %s %q to cluster port group: %q (%v)",
+			lb, aclUUID, aclName, stderr, err)
+		return "", err
 	}
+
+	// Associate ACL UUID with load balancer and ip+port so we can remove this ACL if
+	// backends are re-added.
+	ovn.setServiceACLToLB(lb, vip, aclUUID)
```

Comment:
this needs to go back into the else statement. We shouldn't be adding an ACL to the cache if the OVN ACL was not created.

Reply:
oh I missed the return if theres an error. This is fine.